src/main/java/org/fcrepo/migration/OcflSessionFactoryFactoryBean.java

```diff
@@ -0,0 +1,105 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.fcrepo.migration;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import edu.wisc.library.ocfl.core.OcflRepositoryBuilder;
+import edu.wisc.library.ocfl.core.extension.storage.layout.config.HashedTruncatedNTupleConfig;
+import edu.wisc.library.ocfl.core.path.mapper.LogicalPathMappers;
+import edu.wisc.library.ocfl.core.storage.filesystem.FileSystemOcflStorage;
+import org.apache.commons.lang3.SystemUtils;
+import org.fcrepo.migration.handlers.ocfl.VanillaOcflObjectSessionFactory;
+import org.fcrepo.storage.ocfl.CommitType;
+import org.fcrepo.storage.ocfl.DefaultOcflObjectSessionFactory;
+import org.fcrepo.storage.ocfl.OcflObjectSessionFactory;
+import org.springframework.beans.factory.FactoryBean;
+
+import java.nio.file.Path;
+
+import static com.fasterxml.jackson.databind.SerializationFeature.WRITE_DATES_AS_TIMESTAMPS;
+
+/**
+ * Spring FactoryBean for easy OcflObjectSessionFactory creation.
+ *
+ * @author pwinckles
+ */
+public class OcflSessionFactoryFactoryBean implements FactoryBean<OcflObjectSessionFactory> {
+
+    private final Path ocflRoot;
+    private final Path stagingDir;
+    private final MigrationType migrationType;
+    private final String user;
+    private final String userUri;
+
+    /**
+     * @param ocflRoot OCFL storage root
+     * @param stagingDir OCFL staging dir
+     * @param migrationType migration type
+     * @param user user to add to OCFL versions
+     * @param userUri user's address
+     */
+    public OcflSessionFactoryFactoryBean(final Path ocflRoot,
+                                         final Path stagingDir,
+                                         final MigrationType migrationType,
+                                         final String user,
+                                         final String userUri) {
+        this.ocflRoot = ocflRoot;
+        this.stagingDir = stagingDir;
+        this.migrationType = migrationType;
+        this.user = user;
+        this.userUri = userUri;
+    }
+
+    @Override
+    public OcflObjectSessionFactory getObject() {
+        final var logicalPathMapper = SystemUtils.IS_OS_WINDOWS ?
+                LogicalPathMappers.percentEncodingWindowsMapper() : LogicalPathMappers.percentEncodingLinuxMapper();
+
+        final var ocflRepo =  new OcflRepositoryBuilder()
+                .layoutConfig(new HashedTruncatedNTupleConfig())
+                .logicalPathMapper(logicalPathMapper)
+                .storage(FileSystemOcflStorage.builder().repositoryRoot(ocflRoot).build())
+                .workDir(stagingDir)
+                .buildMutable();
+
+        if (migrationType == MigrationType.F6_OCFL) {
```

Comment:
Do we want to call the migration type "F6_OCFL" thus tying in the version to the migration type? Or should it be something like "FEDORA_OCFL"?

Reply:
Similar to my object about the datastream ids, I personally don't think this name is inappropriate. If there is an F7 and it uses OCFL, what's to say that it uses the same format?

Reply:
That said, I have less of a problem with changing this name than I do with the variable names. The problem with the variable names is that the class is working with two different versions of Fedora, so it makes sense to distinguish between versions.

Reply:
I changed the enum